source/server/drain_manager_impl.cc

```diff
@@ -14,7 +16,34 @@ namespace Server {
 
 DrainManagerImpl::DrainManagerImpl(Instance& server,
                                    envoy::config::listener::v3::Listener::DrainType drain_type)
-    : server_(server), drain_type_(drain_type) {}
+    : server_(server), drain_type_(drain_type), cbs_(server_.dispatcher()) {}
+
+DrainManagerImpl::DrainManagerImpl(Instance& server,
+                                   envoy::config::listener::v3::Listener::DrainType drain_type,
+                                   Event::Dispatcher& dispatcher)
+    : server_(server), drain_type_(drain_type), cbs_(dispatcher) {}
+
+DrainManagerSharedPtr
+DrainManagerImpl::createChildManager(Event::Dispatcher& dispatcher,
+                                     envoy::config::listener::v3::Listener::DrainType drain_type) {
+  auto child = std::make_shared<DrainManagerImpl>(server_, drain_type, dispatcher);
+
+  // Wire up the child so that when the parent starts draining, the child also sees the state-change
+  child_drain_cbs_.push_back(addOnDrainCloseCb(
```

Comment:
Is it possible for child_drain_cbs_ to grow very large over the lifetime of the Envoy process?  Possible example: a proxy that receives periodic config updates that add and remove listeners; each listener seem to have its own drain close cb, so the listener removals would result in an ever increasing list of expired child_drain_cbs_

Reply:
It is definitely possible, especially for the main (server-owned) drain-manager. I'll update to address this.